scipy/stats/_distribution_infrastructure.py

```diff
@@ -3608,25 +3624,11 @@ def _pmf_formula(self, x, **params):
     def _logpmf_formula(self, x, **params):
         return np.full_like(x, -np.inf)
 
-    def _pxf_dispatch(self, x, *, method=None, **params):
-        return self._pdf_dispatch(x, method=method, **params)
-
-    def _logpxf_dispatch(self, x, *, method=None, **params):
-        return self._logpdf_dispatch(x, method=method, **params)
-
 
 class DiscreteDistribution(UnivariateDistribution):
-    def __add__(self, loc):
-        loc = np.asarray(loc)
-        integral = (loc == np.round(loc))
-        loc[~integral] = np.nan
```

Comment:
If non-integral values are going to result in NaN, we'd want to use the parameter infrastructure to deal with this. This suggests the need for different discrete and continuous shifted/scaled distributions -  the continuous ones accept continuous loc/scale parameters, the discrete versions only accept integral ones.